internal/gps/selection.go

```diff
@@ -59,13 +62,43 @@ func (s *selection) popSelection() (atomWithPackages, bool) {
 	return sel.a, sel.first
 }
 
+func (s *selection) findCaseConflicts(pr ProjectRoot) (bool, ProjectRoot) {
+	prlist, has := s.prLenMap[len(pr)]
+	if !has {
+		return false, ""
+	}
+
+	// TODO(sdboyer) bug here if it's possible that strings.ToLower() could
+	// change the length of the string
```

Comment:
Will dropping in `utf8.RuneCountInString()` for `len()` fix this?

What if the map was keyed on the lowercase string rather than their length or rune-count?

Reply:
i believe i considered that approach initially, then dismissed it. lemme reconstruct my thought process...

Reply:
Were you betting on `len()` being a shortcut/hueristic to do less `ToLower` calls overall?
I was thinking that if you can't use `len()` (`O(1)`), and are forced to scan each one to count the runes anyways (`O(n)`), then `ToLower()` isn't any more complex at least. But there might be other things to consider or constant factors dominating.

Reply:
i can't reconstruct the original thought process, but it's fine, because i just bit the bullet and dug into [how the toolchain does this](https://github.com/golang/go/blob/a9216a0ade61ae3c3a45f9cd358a8a38c5d241a5/src/cmd/go/internal/load/pkg.go#L1148-L1157).

They have a special [`ToFold()` helper](https://github.com/golang/go/blob/a9216a0ade61ae3c3a45f9cd358a8a38c5d241a5/src/cmd/go/internal/str/str.go#L32-L69), which generates a stable, folded representation of a string that we can key the map on. at least, i guess that's what it does - i thought [this was harder](http://unicode.org/faq/casemap_charprop.html#2). but, we're at least sorta covered here, because the compiler _is_ the ultimate arbiter of what's acceptable here. of course, i'm relying on current implementation of just one compiler, not a spec, and maybe case-insensitive filesystems will disagree about this definition of case equivalence...but now we're in the domain of some truly far-fetched cases, so whatever.

`ToFold()` is still O(n) even in its fast case, but we no longer need the slice on the other side, so we have fewer checks to do on the other side of the lookup. So, likely still slower than the O(1) `len()`, but at least we get something back. Either way, it's worth it to not have to care about.

Reply:
lololol apparently i found `toFold()` before - https://github.com/golang/dep/issues/433#issuecomment-320141914 - and forgot about it when i actually sat down to implement this